lib/helper.js

```diff
@@ -59,50 +59,53 @@ function loopKeys(obj, fn) {
 }
 module.exports.loopKeys = loopKeys;
 
-function convertPseudoType(obj, options) {
+function convertPseudotype(obj, options) {
+  var reqlType = obj['$reql_type$'];
+  if (reqlType === 'TIME' && options['timeFormat'] !== 'raw') {
+    return new Date(obj['epoch_time'] * 1000);
+  }
+  else if (reqlType === 'GROUPED_DATA' && options['groupFormat'] !== 'raw') {
+    var result = [];
+    for (var i = 0, len = obj['data'].length, ref; i < len; i++) {
+      ref = obj.data[i];
+      result.push({
+        group: ref[0],
```

Comment:
I saw you were doing that, but as I'm calling ```recursivelyConvertPseudotype``` for all the object keys before calling ```convertPseudotype```, I believe it produces the same result.